smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/extension/ChannelProducer.java

```diff
@@ -72,19 +72,51 @@
         return cast(emitter);
     }
 
+    /**
+     * Injection point for emitter using the legacy / deprecated interface.
+     * 
+     * @param injectionPoint the injection point
+     * @param <T> the type
+     * @return the legacy emitter
+     */
+    @Produces
+    @Channel("") // Stream name is ignored during type-safe resolution
+    @Deprecated
+    <T> io.smallrye.reactive.messaging.annotations.Emitter<T> produceEmitterLegacy(InjectionPoint injectionPoint) {
+        LegacyEmitterImpl emitter = new LegacyEmitterImpl(getEmitter(injectionPoint));
+        return cast(emitter);
+    }
+
     @Produces
     @Stream("")
-    <T> Emitter<T> produceEmitterLegacy(InjectionPoint injectionPoint) {
+    @Deprecated
```

Comment:
Should this have `@Deprecated`?

Reply:
I was wondering if Deprecated made sense here as it's very internal. I believe it would help me removing the old emitter once we decide to remove it completely. But no strong opinion.

Reply:
Sorry, was referring to `produceEmiiterStream()` as having `@Deprecated`.

That's the new method right? So was confused as to why it's marked deprecated

Reply:
I've added the javadoc and annotation to all the injection method from ChannelPRoducer explaining what are they injecting and why some of them are deprecated.